ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/ResourceMapperImpl.java

```diff
@@ -174,9 +184,10 @@ private void addPathItem(String httpMethod, Paths path, Operation operation, Str
      *
      * @return Operation Adaptor object of given resource
      */
-    private Optional<OperationInventory> convertResourceToOperation(FunctionDefinitionNode resource, String httpMethod,
-                                                                    String generateRelativePath,
-                                                                    Components components) {
+    private Optional<OperationInventory> convertResourceToOperation(String packageId, SemanticModel semanticModel,
```

Comment:
Passing many arguments to a method can be detrimental to the code's quality. Shall we consider using the Parameter Object pattern[1] or any other suitable approach to enhance code clarity?
[1] https://www.baeldung.com/java-best-practices-many-parameters-method#1-parameter-object

Reply:
Fixed with [63ac241d79644776e53dc2f338009c91970f6df7](https://github.com/ballerina-platform/openapi-tools/pull/1589/commits/63ac241d79644776e53dc2f338009c91970f6df7)